src/dot-merlin/dot_merlin_reader.ml

```diff
@@ -356,7 +356,8 @@ let prepend_config ~cwd ~cfg =
       | Some p ->
         log ~title:"conflicting paths for stdlib" "%s\n%s" p canon_path
       end;
-      { cfg with stdlib = Some canon_path }
+      { cfg with stdlib = Some canon_path;
+                 pass_forward = `STDLIB canon_path :: cfg.pass_forward }
```

Comment:
Is this the correct place to fix the issue ? Why do we need to both update the threaded config and add it to the pass_foward list ?

I would expect the issue being related to some configuration merging that ignores the correct stdlib field in config.

Reply:
Hm I think you're right that this isn't the best way to do it - I've updated the PR.

The issue doesn't seem to be with any configuration merging. I think it is instead that `dot_merlin_reader` never passes on the `STDLIB` directive, so `mconfig_dot` never sees it.